packages/run-protocol/src/runStake/runStakeManager.js

```diff
@@ -6,123 +6,165 @@ import { makeRatio } from '@agoric/zoe/src/contractSupport/ratio.js';
  * @param {ZCF} zcf
  * @param {ZCFMint<'nat'>} debtMint
  * @param {{ debt: Brand<'nat'>, Attestation: Brand<'copyBag'>, Stake: Brand<'nat'> }} brands
- * @param {{
- *  getDebtLimit: () => Amount<'nat'>,
- *  getInterestRate: () => Ratio,
- *  getMintingRatio: () => Ratio,
- *  getLoanFee: () => Ratio,
- * }} paramManager
- * @param {MintAndReallocate} mintAndReallocateWithFee
- * @param {BurnDebt} burnDebt
+ * @param {{ burnDebt: BurnDebt, getGovernedParams: () => ParamManager, mintAndReallocate: MintAndReallocate }} mintPowers
  * @param {Object} timing
  * @param {ERef<TimerService>} timing.timerService
  * @param {bigint} timing.chargingPeriod
  * @param {bigint} timing.recordingPeriod
  * @param {bigint} timing.startTimeStamp
  *
- * @typedef {ReturnType<typeof makeRunStakeManager>} RunStakeManager
+ * @returns {State}
  */
-export const makeRunStakeManager = (
+const initState = (
   zcf,
   debtMint,
   brands,
-  paramManager,
-  mintAndReallocateWithFee,
-  burnDebt,
-  { timerService, chargingPeriod, recordingPeriod, startTimeStamp },
+  mintPowers,
+  { chargingPeriod, recordingPeriod, startTimeStamp, timerService },
 ) => {
-  /** @param { Amount<'copyBag'>} attestationGiven */
-  const maxDebtForLien = attestationGiven => {
-    const mintingRatio = paramManager.getMintingRatio();
-    assert.equal(
-      mintingRatio.numerator.brand,
-      brands.debt,
-      X`${mintingRatio} not in Debt / Stake`,
-    );
-    assert.equal(
-      mintingRatio.denominator.brand,
-      brands.Stake,
-      X`${mintingRatio} not in Debt / Stake`,
-    );
-    assert.equal(
-      attestationGiven.brand,
-      brands.Attestation,
-      X`Invalid Attestation ${attestationGiven}. Expected brand ${brands.Attestation}`,
-    );
-    fit(attestationGiven.value, M.bagOf([M.string(), M.bigint()]));
-    const [[_addr, valueLiened]] = getCopyBagEntries(attestationGiven.value);
-    const amountLiened = AmountMath.make(brands.Stake, valueLiened);
-    const maxDebt = floorMultiplyBy(amountLiened, mintingRatio);
-    return { maxDebt, amountLiened };
-  };
+  const totalDebt = AmountMath.makeEmpty(brands.debt, 'nat');
+  const compoundedInterest = makeRatio(100n, brands.debt); // starts at 1.0, no interest
+  const latestInterestUpdate = startTimeStamp;
 
-  let totalDebt = AmountMath.makeEmpty(brands.debt, 'nat');
-  let compoundedInterest = makeRatio(100n, brands.debt); // starts at 1.0, no interest
-  let latestInterestUpdate = startTimeStamp;
-  const { updater: assetUpdater, notifier: assetNotifer } = makeNotifierKit(
+  const { updater: assetUpdater, notifier: assetNotifier } = makeNotifierKit(
     harden({
       compoundedInterest,
-      interestRate: paramManager.getInterestRate(),
+      interestRate: mintPowers.getGovernedParams().getInterestRate(),
       latestInterestUpdate,
       totalDebt,
     }),
   );
 
+  // ??? does this promise need to be a presence?
+  const periodNotifier = E(timerService).makeNotifier(0n, recordingPeriod);
```

Comment:
Yes, if you store the `periodNotifier` Promise in `state`, then 1: it will trigger the #5106 bug that I haven't fixed yet, and 2: it won't be durable (once you change `defineKind` to `defineDurableKind`, your `init` calls will start failing with a complaint about a non-durable value (a Promise) being stored in `state`).

Also, it doesn't look like this notifier needs to be available to the main methods.. it's only used during `finish` to wire up a responder. See my comment below on another approach for that.